webapp/channels/src/components/menu/menu_item.tsx

```diff
@@ -79,6 +79,18 @@ export interface Props extends MuiMenuItemProps {
 
     onClick?: (event: MouseEvent<HTMLLIElement> | KeyboardEvent<HTMLLIElement>) => void;
 
+    /**
+     * If true, clicking on the menu item will not close the menu.
+     * @warning This is a rare UX pattern and should be used sparingly as it breaks a lot of menu/submenu auto closing logic.
+     */
+    antipattern__blockClosingOnClick?: boolean;
```

Comment:
![Untitled](https://github.com/mattermost/mattermost/assets/17708702/34c97fb3-211e-4c4a-8011-8a1e8248e377)

1. Because as per Aria guidelines the menu items should close the menu on click
1. That logic is in place to avoid the problem with react bootstrap modal interfereing and infinite recursion
1. That indeed is anti pattern
1. For filters please see the image above

Reply:
So the fact that this is labelled `antipattern` is a huge code smell to me here, meaning that I think we might be misusing this component then. I know we have components like `react-select` that might be better suited here instead of trying to use the `Menu` component which is supposed to behave like a menu.

I'm thinking we need to come up with a different solution, and maybe that is a different component then. If we're adamant on keeping the functionality of the Menu correct (which makes sense) then I think that's the way forward.

Reply:
React select which implements a <select> is not used in this purpose, since this is a menu using the Menu component is the correct approach.

<img width="945" alt="Screenshot 2024-01-17 at 18 59 41" src="https://github.com/mattermost/mattermost/assets/17708702/beb2ae90-bbd2-4c30-b93a-207ee8bd5fa7">
 
However i read bit more about it and found the above in the specs, https://www.w3.org/WAI/ARIA/apg/patterns/menubar/

So i implemented that above

Reply:
React select which implements a <select> is not used in this purpose, since this is a menu using the Menu component is the correct approach.

> 
    (Optional): When focus is on a menuitemcheckbox, changes the state without closing the menu.
    (Optional): When focus is on a menuitemradio that is not checked, without closing the menu, checks the focused menuitemradio and unchecks any other checked menuitemradio element in the same group.
    (Optional): When focus is on a menuitem that has a submenu, opens the submenu and places focus on its first item.
    (Optional): When focus is on a menuitem that does not have a submenu, activates the menuitem and closes the menu.


However i read bit more about it and found the above in the specs, https://www.w3.org/WAI/ARIA/apg/patterns/menubar/

So i implemented that above

Reply:
Yep that looks much more concise :)